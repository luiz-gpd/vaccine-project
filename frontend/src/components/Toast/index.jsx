import React, { useContext } from 'react'
import AppContext from '../../AppContext'
import ClayAlert from '@clayui/alert'

const Toast = (children, title, type) => {
    
    const {toast, setToast} = useContext(AppContext);
    
    //  {setToastItems([...toastItems, Math.random() * 100])}
    return (
        <ClayAlert.ToastContainer>
            {toast.map(value => (
                <ClayAlert
                    autoClose={5000}
                    key={value}
                    onClose={() => {
                        setToast(prevItems =>
                            prevItems.filter(item => item !== value)
                        );
                    }}
                    className={type}
                    title={title}
                >{children}</ClayAlert>
            ))}
        </ClayAlert.ToastContainer>


    )
}
export default Toast