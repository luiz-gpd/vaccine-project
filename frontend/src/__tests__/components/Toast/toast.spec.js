import Toast from "../../../components/Toast"
import { render } from "@testing-library/react"

describe("Toast component", () => {
    it("renders", () => {
        const { asFragment } = render(
            <Toast
            toast={[1000]}
          >
          </Toast>);

        expect(asFragment()).toMatchSnapshot();
    });
    it("renders with props", () => {
    
        const { queryByText } = render(
          <Toast
            title="Título"
            type="success"
            toast={[1000]}
          >
              <div className="information">
                    <span>Uno</span>
                    <span>Due</span>
                    <span>Tre</span>
                </div>
          </Toast>
        );
        expect(queryByText("Título")).toBeTruthy();
        expect(queryByText("Uno")).toBeTruthy();
        expect(queryByText("Due")).toBeTruthy();
        expect(queryByText("Tre")).toBeTruthy();
    });
});