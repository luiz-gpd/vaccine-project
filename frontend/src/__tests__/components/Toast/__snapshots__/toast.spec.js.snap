// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast component renders 1`] = `
<DocumentFragment>
  <div
    class="alert-container container"
  >
    <div
      class="alert-notifications alert-notifications-fixed"
    >
      <div
        class="alert alert-info"
        role="alert"
      >
        <div
          class="alert-autofit-row autofit-row"
        >
          <div
            class="autofit-col"
          >
            <div
              class="autofit-section"
            >
              <span
                class="alert-indicator"
              >
                <svg
                  class="lexicon-icon lexicon-icon-info-circle"
                  role="presentation"
                >
                  <use
                    xlink:href="#info-circle"
                  />
                </svg>
              </span>
            </div>
          </div>
          <div
            class="autofit-col autofit-col-expand"
          >
            <div
              class="autofit-section"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
